src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -265,7 +259,7 @@ def filtered_states_generator(
                 _linearise_update_at=_linearise_update_at,
             )
 
-            yield filtrv, info_dict
+            yield times[idx], filtrv, info_dict
```

Comment:
sorry if this has been explained before.
Can you please remind me what the reason for returning the time value here was? It seems like this value would be accessible outside of the generator, but I might be missing something.

Reply:
It is returned such that when building the `FilteringPosterior` on the fly, we can call `posterior.append(location=t, state=rv)`.
Do you think there's a more elegant solution?

Reply:
not sure. I guess the alternative would be to include zip over `regression_problem.locations, filter_generator(regression_problem)` but (a) this looks awkward, and (b) the current implementation *might* be more robust w.r.t. multiple measurements at a single locations.